driver/grpc.go

```diff
@@ -144,29 +144,43 @@ func (c *grpcDriver) dropDatabaseWithOptions(ctx context.Context, db string, opt
 }
 
 func (c *grpcDriver) beginTxWithOptions(ctx context.Context, db string, options *TxOptions) (txWithOptions, error) {
+	var respHeaders grpcmd.MD // variable to store header and trailer
 	resp, err := c.api.BeginTransaction(ctx, &api.BeginTransactionRequest{
 		Db:      db,
 		Options: (*api.TransactionOptions)(options),
-	})
+	}, grpc.Header(&respHeaders))
+
 	if err != nil {
 		return nil, GRPCError(err)
 	}
 	if resp.GetTxCtx() == nil {
 		return nil, GRPCError(fmt.Errorf("empty transaction context in response"))
 	}
-	return &grpcCRUD{db: db, api: c.api, txCtx: resp.GetTxCtx()}, nil
+
+	additionalHeaders := grpcmd.New(map[string]string{})
+	if respHeaders.Get(SetCookieHeaderKey) != nil {
+		for _, incomingCookie := range respHeaders.Get(SetCookieHeaderKey) {
+			additionalHeaders = grpcmd.Join(additionalHeaders, grpcmd.Pairs(CookieHeaderKey, incomingCookie))
+		}
+	}
+
+	return &grpcCRUD{db: db, api: c.api, txCtx: resp.GetTxCtx(), additionalMetadata: additionalHeaders}, nil
 }
 
-func setGRPCTxCtx(ctx context.Context, txCtx *api.TransactionCtx) context.Context {
-	if txCtx == nil || txCtx.Id == "" {
+func setGRPCTxCtx(ctx context.Context, txCtx *api.TransactionCtx, additionalMetadata grpcmd.MD) context.Context {
+	if txCtx == nil || txCtx.Id == "" || additionalMetadata == nil {
```

Comment:
I think we don't need `|| additionalMetadata == nil` check here

Reply:
Fixed.